### PR TITLE
Profile style

### PR DESCRIPTION
**Pull request by jdaniel0x50**
Related to edit profile ticket
Note: This does NOT complete the ticket requirements. It is an initial iterative push to clean up the profile page styles so that Jenna (also working on the profile page) can incorporate these changes as she edits the sidebar.

**Changes in this PR include:**

- Adapted styles to make the profile page and components more responsive to different screen size
- Changed styles to bootstrap styles - and removed custom styles (I am still working on this - this is an initial step in this direction)
- Changed custom input fields to incorporate pre-built input components to handle validation (note - a future build will incorporate validation logic on the form so that invalid inputs will not permit the user to submit the form)

**This PR does not include**:

- Form validation (forbid incomplete or invalid data to submit) (it does, however, include individual input field validation)
- Form fields will default to show current data ... will be an easy fix, but need to incorporate
- Submits will only update fields that change
- Provide a confirmation to the user when data updates successfully
- Ability to see multiple children applications and status (on camp profile)

EDITS PUSHED ON FRIDAY, 8/3/2018

- Added form validation for name, email, and password fields
- Added validation to validate input fields and their confirm fields match
- Disabled submit if form fields are invalid or if confirm fields do not match
- Modified error messages
- NOTE: The form does not currently submit to the back end -- will need to modify the back-end routes and adjust this axios.post in a future PR

GLOBAL EDITS

- added a global cursor: pointer for all buttons and a tags in App.vue
- added a String.trim() function to the input component such that all inputs will not submit with leading or trailing spaces

### CODE REVIEW

File: src/components/forms/inputFieldOnly.vue [hunk 1]
```diff
@@ -309,11 +334,17 @@ export default {
 		},
 		lostFocus: function() {
 			// run validators immediately on losing focus
+			if (typeof this.input === 'string') {
+				// trim string input values
+				this.input = this.input.trim();
+			}
 			this.isPristine = false;
-			this.runValidators();
+			if (this.readonly == false) {
```
Comment (mkarazin) [inline src/components/forms/inputFieldOnly.vue]: I'd recommend use of the identity check `===` over the use of equality `==`, for this and all other conditional checks.

Comment (jdaniel0x50) [reply src/components/forms/inputFieldOnly.vue]: Sounds great. Made the changes and pushed to this PR

**Comment by jdaniel0x50** [comment]
UPDATE on Wednesday, August 8:

Added changes to include a list of all children that have applications on the parent profile. The children.vue sub component uses a new backend API route to group and sort all applications related to the logged in user profile by child name

The list of children, includes the child age and all associated applications for the child - including the application date and applicaiton status

There is also a link on the application date to view the submitted application

image

File: src/components/applications/children.vue [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            applications: [],
+            campApps: [],
+            children: [],
+        }
+    },
+    created: function() {
+        localforage.getItem("X_TOKEN")
+        .then(session => {
+            this.sessionId = session;
+            axios.get("/api/v1/profile/" + session + "/applications", {
```
Comment (mkirlin) [inline src/components/applications/children.vue]: Hey, I altered the back-end pretty significantly. I implemented your changes from your last PR, but I tweaked some pieces of it. You can replace the url here with this: `"/api/v1/profile/applications"` and it should return the same thing.

**Review by mkirlin — CHANGES_REQUESTED** [review]

**Comment by jdaniel0x50** [comment]
I submitted a new revision to this PR this evening, including the following changes:

- Addressed the changes requested from Michael (re: the backend API route)
- Added address view and edit fields to the profile page (re: conversation on Slack) / note this does not change the profile address information on registration - this will be submitted in a separate PR
- Added email validation to the input component
- Divided the large profile component into three components: profile, profile view, profile edit

**Review by mkirlin — APPROVED** [review]

**Review by zachgarwood — APPROVED** [review]
LGTM 🦄

**Pull request merged**